src/graph/neighbors.h

```diff
@@ -0,0 +1,36 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef IGRAPH_NEIGHBORS_H
+#define IGRAPH_NEIGHBORS_H
+
```

Comment:
Ideally, the header should stand on its own so we should `#include` all the headers here that define `igraph_t`, `igraph_vector_t`, `igraph_integer_t`, `igraph_neimode_t` and all the other igraph stuff that  you use in the header.